examples/indexing_routines.py

```diff
@@ -0,0 +1,302 @@
+#!/usr/bin/env python
+
+# Copyright 2021 NVIDIA Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from __future__ import print_function
+
+import argparse
+import datetime
+import gc
+import math
+
+from benchmark import run_benchmark
+from legate.timing import time
+
+import cunumeric as np
+
+
+def compute_diagonal(steps, N, timing, warmup):
+    A1 = np.ones((N,), dtype=int)
+    print("measuring diagonal")
+    for step in range(steps + warmup):
+        if step == warmup:
+            start = time()
+        A2 = np.diag(A1)
+        A1 = np.diag(A2)
+    stop = time()
+    total = (stop - start) / 1000.0
+    if timing:
+        space = (N * N + N) * np.dtype(int).itemsize / 1073741824
+        print("Total Size:       " + str(space) + " GB")
+        print("Elapsed Time for diagonal: " + str(total) + " ms")
+        average = total / steps
+        print("Average            :     " + str(average) + " ms")
+        print(
+            "bandwidth:               "
+            + str(space * 1000.0 / (average))
+            + " GB/s"
+        )
+    return total
+
+
+def compute_choose(steps, N, timing, warmup):
+    print("measuring choose")
+    A = tuple(
+        np.full(
+            (
+                10,
+                N,
+            ),
```

Comment:
Do you want each of these 10 arrays to be 10xN? I thought they should all be the same shape as C1 (i.e. `(N,)`).

Reply:
It will work either way, but I think we want to measure performance when it is (N,) shape